server/controller/todosController.js

```diff
@@ -0,0 +1,138 @@
+const conn = require("../mariadb");
+const { StatusCodes } = require("http-status-codes");
+
+const getTodos = (req, res) => {
+  //user_id는 users랑 합칠때 authorization.id로 변경될 것
+  const { user_id } = req.body;
+  const { today, month } = req.query;
+
+  let sql = `SELECT * FROM todos WHERE user_id = ? `;
+  let values = [];
+
+  if (today) {
+    values.push(user_id, today);
+    sql += `AND due_date > ?`;
+  } else if (month) {
+    values.push(user_id, month + "%");
+    sql += `AND due_date LIKE ?`;
+  } else {
+    values = [user_id];
+  }
+
+  conn.query(sql, values, (err, results) => {
+    if (err) {
+      return res.status(StatusCodes.BAD_REQUEST).end();
+    }
+
+    if (results.length > 0) {
+      res.status(StatusCodes.OK).json(results);
+    } else {
+      res.status(StatusCodes.NOT_FOUND).json(results);
+    }
+  });
+};
+
+const getTodoDetail = (req, res) => {
+  const { id } = req.params;
+
+  const values = [id];
+  const sql = `SELECT * FROM todos WHERE id = ?`;
+
+  conn.query(sql, values, (err, results) => {
+    if (err) {
+      return res.status(StatusCodes.BAD_REQUEST).end();
+    }
+
+    if (results.length > 0) {
+      res.status(StatusCodes.OK).json(results);
+    }
+  });
+};
+
+const addTodos = (req, res) => {
+  //user_id는 users랑 합칠때 authorization.id로 변경될 것
+  const { title, detail, due_date, user_id } = req.body;
+
+  let sql = `INSERT INTO todos (title, detail, due_date, user_id) VALUES(?, ?, ?, ?)`;
+  let values = [title, detail, due_date, user_id];
+
+  if (detail === undefined) {
+    sql = `INSERT INTO todos (title, due_date, user_id) VALUES(?, ?, ?)`;
+    values = [title, due_date, user_id];
+  }
+
+  conn.query(sql, values, (err, results) => {
+    if (err) {
+      console.log(err);
+      return res.status(StatusCodes.BAD_REQUEST).end();
+    }
+    if (results.affectedRows === 0) {
+      return res.status(StatusCodes.BAD_REQUEST).end();
+    } else {
+      return res.status(StatusCodes.CREATED).json(results);
+    }
+  });
+};
+
+const editTodos = (req, res) => {
```

Comment:
update라는 표현을 일반적으로 많이 써서 아래와 같이 표현해줘도 좋을 것 같아요!
```suggestion
const updateTodos = (req, res) => {
```